cdap-ui/app/directives/tooltip-toggle/tooltip-toggle.js

```diff
@@ -39,20 +39,22 @@ angular.module(PKG.name + '.commons')
     return {
       restrict: 'A',
       scope: {
-        ellipsis: '='
+        ellipsis: '=',
+        offsetValue: '='
       },
-      link: function (scope, element, attrs) {
+      link: function (scope, element) {
         function isEllipsisActive(e) {
-          return (e[0].offsetWidth > e[0].parentElement.offsetWidth - 20);
+          return (e[0].offsetWidth > e[0].parentElement.offsetWidth - scope.offsetValue);
         }
 
-        scope.$watch(attrs.tooltipEllipsis, function () {
-          if (attrs.tooltipEllipsis) {
-            if (isEllipsisActive(element)) {
-              scope.ellipsis = true;
-            } else {
-              scope.ellipsis = false;
-            }
+        scope.$watch(function () {
```

Comment:
This might also be a little less performance efficient way. We already have lots of graphs that we display and lots of DOM operations (hovers, clicks on graphs) and on top of that a ```$watch``` on each tab title. We might want to make this better.